### PR TITLE
Migrate JPA Spec 2.0 DerivedIdentity/Util to OL

### PR DESCRIPTION
**Pull request by jgrassel**
Migrate JPA 2.0 Spec buckets Util and Derived Identity to Open Liberty, and rename the JPA Spec 2.0 fat project to `com.ibm.ws.jpa.tests.spec20_fat`

To reduce increased exec time due to enabling test containers, broke out several sub-buckets into their own test projects:
```
com.ibm.ws.jpa.tests.spec20.entitymanager_fat
com.ibm.ws.jpa.tests.spec20.ordercolumn_fat
com.ibm.ws.jpa.tests.spec20.querylockmode_fat
com.ibm.ws.jpa.tests.spec20_fat
```

### CODE REVIEW

**Comment by jgrassel** [comment]
#build

File: dev/com.ibm.ws.jpa.tests.spec20_fat/fat/src/com/ibm/ws/jpa/spec20/JPA20DerivedIdentity_WEB.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.jpa.spec20;
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.spec20_fat/fat/src/com/ibm/ws/jpa/spec20/JPA20DerivedIdentity_WEB.java]: It looks like the package name is still `com.ibm.ws.jpa.spec20`. Can we get this update to `com.ibm.ws.jpa.tests.spec20` to match the new project name?

File: dev/com.ibm.ws.jpa.tests.spec20_fat/publish/servers/JPA20UtilEJBServer/server.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            <jdbcDriver libraryRef="JDBCLib"/>
+            <properties.derby.embedded databaseName="memory:ds1" createDatabase="create" user="dbuser1" password="{xor}Oz0vKDtu" />
+    </dataSource>
+    
+    <dataSource id="jdbc/JPA_NJTADS" jndiName="jdbc/JPA_NJTADS" fat.modify="true" transactional="false">
+            <jdbcDriver libraryRef="JDBCLib"/>
+            <properties.derby.embedded databaseName="memory:ds1" createDatabase="create" user="dbuser1" password="{xor}Oz0vKDtu" />
+    </dataSource>
+    
+    <library id="JDBCLib" fat.modify="true" apiTypeVisibility="+third-party">
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.spec20_fat/publish/servers/JPA20UtilEJBServer/server.xml]: Is the thirdparty visibility necessary? This probably should update to use the format of `AnonymousJDBCLib` most of our other FAT use (see `com.ibm.ws.jpa.tests.spec10.inheritance_fat.common\publish\servers\JPA10InheritanceEJBServer\server.xml` or `com.ibm.ws.jpa.tests.spec10.query_fat.common\publish\servers\JPA10QueryServer\server.xml`)

File: dev/com.ibm.ws.jpa.tests.spec20_fat/publish/servers/JPA20Server/bootstrap.properties [hunk 3]
```diff
@@ -0,0 +1,11 @@
+###############################################################################
+# Copyright (c) 2017 IBM Corporation and others.
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.spec20_fat/publish/servers/JPA20Server/bootstrap.properties:2]: Maybe do a search/replace for "2017 IBM Corporation" -> "2021 IBM Corporation"

**Review by dazey3 — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.jpa.tests.spec20_fat/fat/src/com/ibm/ws/jpa/spec20/JPA20DerivedIdentity_WEB.java (hunk 1 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.spec20_fat/fat/src/com/ibm/ws/jpa/spec20/JPA20DerivedIdentity_WEB.java]: Yep, I'll do that.

File: dev/com.ibm.ws.jpa.tests.spec20_fat/publish/servers/JPA20UtilEJBServer/server.xml (hunk 2 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.spec20_fat/publish/servers/JPA20UtilEJBServer/server.xml]: Good catch, fixing that too.

File: dev/com.ibm.ws.jpa.tests.spec20_fat/publish/servers/JPA20Server/bootstrap.properties (hunk 3 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.spec20_fat/publish/servers/JPA20Server/bootstrap.properties:2]: Hmm, only if there's  changes inside the files.  I don't think moving a file is enough to change its copyright date.

**Comment by jgrassel** [comment]
#build
#run-libby-bot

**Comment by jgrassel** [comment]
#run-libby-bot  - forgot to update JPA_2 to include the breakout jpa spec20 buckets.

File: dev/com.ibm.ws.jpa.tests.spec20.entitymanager_fat/build.gradle [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.spec20.entitymanager_fat/build.gradle]: Copyright (c) 2021 IBM Corporation and others.

File: dev/com.ibm.ws.jpa.tests.spec20.entitymanager_fat/fat/src/com/ibm/ws/jpa/tests/spec20/entitymanager/FATSuite.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.spec20.entitymanager_fat/fat/src/com/ibm/ws/jpa/tests/spec20/entitymanager/FATSuite.java]: Copyright (c) 2021 IBM Corporation and others.

**Comment by jgrassel** [comment]
#run-libby-bot

**Comment by jgrassel** [comment]
#run-libby-bot. updated copyright dates on files.

File: dev/com.ibm.ws.jpa.tests.spec20.entitymanager_fat/test-applications/entitymanager/src/com/ibm/ws/jpa/entitymanager/model/JPA20EntityManagerDetachEntity.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
+package com.ibm.ws.jpa.entitymanager.model;
```
Comment (dazey3) [inline dev/com.ibm.ws.jpa.tests.spec20.entitymanager_fat/test-applications/entitymanager/src/com/ibm/ws/jpa/entitymanager/model/JPA20EntityManagerDetachEntity.java:12]: Need a header on this file

**Review by dazey3 — CHANGES_REQUESTED** [review]
small fix to the existing Entity class being moved

File: dev/com.ibm.ws.jpa.tests.spec20.entitymanager_fat/test-applications/entitymanager/src/com/ibm/ws/jpa/entitymanager/model/JPA20EntityManagerDetachEntity.java (hunk 6 above)
Comment (jgrassel) [reply dev/com.ibm.ws.jpa.tests.spec20.entitymanager_fat/test-applications/entitymanager/src/com/ibm/ws/jpa/entitymanager/model/JPA20EntityManagerDetachEntity.java:12]: Fixed

**Comment by jgrassel** [comment]
#run-libby-bot
#build

**Review by dazey3 — APPROVED** [review]
LGTM

**Pull request merged**